src/filesystem.cpp

```diff
@@ -245,12 +245,12 @@ namespace fc {
          boost::filesystem::copy( boost::filesystem::path(f), boost::filesystem::path(t), ec );
          if( ec )
          {
-            FC_THROW( "Copy from ${srcfile} to ${dstfile} failed because ${reason}:${value}",
-                      ("srcfile",f)("dstfile",t)("reason",ec.category().name())("value",ec.value()) );
+            FC_THROW( "Copy from ${srcfile} to ${dstfile} failed because ${code} : ${message}",
+                      ("srcfile",f)("dstfile",t)("code",ec.value())("message",ec.message()) );
          }
      } catch ( boost::system::system_error& e ) {
      	FC_THROW( "Copy from ${srcfile} to ${dstfile} failed because ${reason}",
-	         ("srcfile",f)("dstfile",t)("reason",e.what() ) );
+	         ("srcfile",f)("dstfile",t)("reason",std::string(e.what()) ) );
```

Comment:
I do not want to get into a long discussion about this, but I do not see how the c-string returned by e.what() has an opportunity to go out of scope before FC_THROW returns (barring threads, which I do not think are in play here). Therefore, I see the std::string(e.what()) as useless.

Now, I can see how using c-strings can cause scope issues before we enter the catch. But wrapping e.what() in std::string is closing the barn door after the cattle have left (see https://pastebin.com/nkKL8W9n ).

Someone please help me see the error in my thinking.